### PR TITLE
[feature](Nereids): when cost time > 5s, throw timeout Exception

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem summary

when cost time > 10s, throw timeout Exception

## Checklist(Required)

* [x] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/scheduler/SimpleJobScheduler.java [hunk 1]
```diff
@@ -27,6 +32,12 @@ public class SimpleJobScheduler implements JobScheduler {
     public void executeJobPool(ScheduleContext scheduleContext) {
         JobPool pool = scheduleContext.getJobPool();
         while (!pool.isEmpty()) {
+            CascadesContext context = (CascadesContext) scheduleContext;
+            Stopwatch watch = context.getStopwatch();
+            // when cost time > 10s. we will stop this job.
+            if (watch.elapsed(TimeUnit.MILLISECONDS) > 10000) {
+                throw new RuntimeException("Nereids cost too much time ( > 10s )");
+            }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/scheduler/SimpleJobScheduler.java:40]: the StopWatch's doc say
```
Note that the overhead of measurement can be more than a microsecond
```
so, if we have more than 1000 jobs, then the overhead will more than 1 second, and it is unacceptable

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildp0

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildp0

**Review by morrySnow — APPROVED** [review]

**Pull request merged**